Sources/Extensions/UIKit/UIAlertControllerExtensions.swift

```diff
@@ -17,20 +17,74 @@ import AudioToolbox
-	public func show(animated: Bool = true, vibrate: Bool = false, completion: (() -> Void)? = nil) {
-		UIApplication.shared.keyWindow?.rootViewController?.present(self, animated: animated, completion: completion)
-		if vibrate {
-			#if canImport(AudioToolbox)
-			AudioServicesPlayAlertSound(kSystemSoundID_Vibrate)
-			#endif
-		}
-	}
+  // swiftlint:disable next cyclomatic_complexity
+  /// SwifterSwift: Present alert view controller.
+  ///
+  /// - Parameters:
+  ///   - viewController: a custom view controller to be presented (default is the rootViewController of application's key window).
+  ///   - anchor: specify the anchor for the popover’s arrow points to (default is nil).
+  ///   - arrowDirection: specify the permitted arrow directions for the popover (default is UIPopoverArrowDirection.any).
+  ///   - animated: set true to animate presentation of alert controller (default is true).
+  ///   - vibrate: set true to vibrate the device while presenting the alert (default is false).
+  ///   - completion: an optional completion handler to be called after presenting alert controller (default is nil).
+  public func show(in viewController: UIViewController? = UIApplication.shared.keyWindow?.rootViewController, anchor: AnyObject? = nil, arrowDirection: UIPopoverArrowDirection = .any, animated: Bool = true, vibrate: Bool = false, completion: (() -> Void)? = nil) {
+    // https://github.com/steipete/PSTAlertController/blob/master/PSTAlertController/PSTAlertController.m#L278
+    // We absolutely need a controller going forward.
+    guard var viewController = viewController else {
+      assertionFailure("Can't show alert because there is no root view controller.")
+      return
+    }
+
+    // Use the frontmost viewController for presentation.
+    while let frontViewController = viewController.presentedViewController, !frontViewController.isBeingDismissed {
+      viewController = frontViewController
+    }
+
+    // If it has already been set, we should not overwrite it. Will do nothing on iPhone since the popover is nil.
+    if let popover = popoverPresentationController, popover.sourceView == nil, popover.barButtonItem == nil {
+      switch anchor {
+      case let value as UIBarButtonItem:
+        popover.barButtonItem = value
+      case let value as UIView:
+        popover.sourceView = value
+        popover.sourceRect = value.bounds
+      case let value as NSValue:
+        popover.sourceView = viewController.view
+        popover.sourceRect = value.cgRectValue
+      default:
+        popover.sourceView = viewController.view
+        popover.sourceRect = viewController.view.bounds
+      }
+
+      // Workaround for rdar://18921595. Unsatisfiable constraints when presenting UIAlertController.
+      // If the rect is too large, the action sheet can't be displayed.
+      let sourceRect: CGRect = popover.sourceRect
```

Comment:
You don't need to define the types, they will be easily inferred by the compiler.